packages/medusa/src/utils/get-product-pricing-with-pricing-module.ts

```diff
@@ -0,0 +1,124 @@
+import { Request } from "express"
+import { PriceSelectionContext } from "../interfaces"
+import { MoneyAmount, ProductVariant } from "../models"
+import { isDefined } from "medusa-core-utils"
+import { ProductVariantPricing } from "../types/pricing"
+
+export async function getProductPricingWithPricingModule(
+  req: Request,
+  variants: ProductVariant[],
+  context: PriceSelectionContext = {}
+) {
+  const remoteQuery = req.scope.resolve("remoteQuery")
+  const pricingModule = req.scope.resolve("pricingModuleService")
+
+  const variables = {
+    variant_id: variants.map((v) => v.id),
+  }
+
+  const priceSetQuery = `
+    query {
+      variants {
+        id
+        title
+        price {
+          price_set {
+            id
+          }
+        }
+      }
+    }
+   `
+
+  const res: VariantsRes[] = await remoteQuery(priceSetQuery, {
+    context: variables,
+  })
+
+  const variantIdToPriceSetIdsMap = new Map(
+    res.map((r) => {
+      if (!r.price) {
+        return [r.id, []]
+      }
+
+      const value = Array.isArray(r.price)
+        ? r.price.map((p) => p.price_set_id)
+        : [r.price.price_set_id]
+      return [r.id, value]
+    })
+  )
+
+  const priceSetIds = res
+    .map((r) => {
```

Comment:
Todo: please can we stop the single letter naming. What is r here since it comes from res, what is res. As a reader you need to go back up to be sure. We agreed last year that single letter was reserved for inline function, and only if the variable before is understandable.